File: src/main/java/com/example/mandatorytwo/models/Champion.java
package com.example.mandatorytwo.models;


import lombok.Data;
import lombok.Generated;

import javax.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.Set;

@Data
@Table(name="Champions")
@Entity
public class Champion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private Long championId;

    @Column
    private String name;

    @Enumerated(value = EnumType.STRING)
    @Column
    private Role role;

    @Enumerated(value = EnumType.STRING)
    @Column
    private Difficulty difficulty;

    @Column(length = 2000)
    private String lore;

    @Column(length = 10000)
    private String image;

    @JsonIgnore
    @OneToMany(mappedBy = "champion", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Set<Contestant> contestants;

}

File: src/main/resources/templates/champions/champion.js
const queryString = window.location.search;
const URLParams = new URLSearchParams(queryString);
const championId = URLParams.get("championId");


fetch("http://localhost:8080/champions/" + championId)
    .then(response => response.json())
    .then(champion => {
        console.log(champion);
        document.getElementById("champion-name").innerText = champion.name;
        document.getElementById("champion-lore").innerText = champion.lore;
        document.getElementById("champion-role").innerText = champion.role;
        document.getElementById("champion-difficulty").innerText = champion.difficulty;

    });